src/Aspire.Hosting.Analyzers/AppHostAnalyzer.Diagnostics.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+
+namespace Aspire.Hosting.Analyzers;
+
+public partial class AppHostAnalyzer
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("MicrosoftCodeAnalysisReleaseTracking", "RS2008:Enable analyzer release tracking")]
+    internal static class Diagnostics
+    {
+        internal static readonly DiagnosticDescriptor s_modelNameMustBeValid = new(
+            id: "ASPIRE0000",
```

Comment:
We already have

* ASPIRE001
* ASPIRE002
* ...
* ASPIRE005

We should probably rectify these warning IDs. It might make sense to put them in a shared place so we can see them and know what each of them are.

Example: https://github.com/dotnet/runtime/blob/main/docs/project/list-of-diagnostics.md

Reply:
Only 3 digits? Man we better use them sparingly I guess!

Reply:
Added .\docs\list-of-diagnostics.md

Reply:
Should I change all of them to use the more typical four-digit format?

Reply:
> Only 3 digits?

Can always add `ASPIRE1001` in the future.

> Should I change all of them to use the more typical four-digit format?

I believe that would be a breaking change. Probably not worth it.